src/overlays/actors/ovl_Boss_Ganondrof/z_boss_ganondrof.c

```diff
@@ -1,15 +1,102 @@
+/*
+ * File: z_boss_ganondrof.c
+ * Overlay: ovl_Boss_Ganondrof
+ * Description: Phantom Ganon
+ */
+
 #include "z_boss_ganondrof.h"
+#include "objects/object_gnd/object_gnd.h"
+#include "overlays/actors/ovl_En_fHG/z_en_fhg.h"
+#include "overlays/actors/ovl_En_Fhg_Fire/z_en_fhg_fire.h"
+#include "overlays/effects/ovl_Effect_Ss_Fhg_Flash/z_eff_ss_fhg_flash.h"
+#include "overlays/effects/ovl_Effect_Ss_Hahen/z_eff_ss_hahen.h"
 
 #define FLAGS 0x00000035
 
 #define THIS ((BossGanondrof*)thisx)
 
+typedef enum {
+    /* 0 */ NOT_DEAD,
+    /* 1 */ DEATH_START,
+    /* 2 */ DEATH_THROE,
+    /* 3 */ DEATH_WARP,
+    /* 4 */ DEATH_SCREAM,
+    /* 5 */ DEATH_DISINTEGRATE,
+    /* 6 */ DEATH_FINISH
+} BossGanondrofDeathStates;
+
+typedef enum {
+    /* 0 */ THROW_NORMAL,
+    /* 1 */ THROW_SLOW
+} BossGanondrofThrowActions;
+
+typedef enum {
+    /* 0 */ STUNNED_FALL,
+    /* 1 */ STUNNED_GROUND
+} BossGanondrofStunnedActions;
+
+typedef enum {
+    /* 0 */ CHARGE_WINDUP,
+    /* 1 */ CHARGE_START,
+    /* 2 */ CHARGE_ATTACK,
+    /* 3 */ CHARGE_FINISH
+} BossGanondrofChargeActions;
+
+typedef enum {
+    /* 0 */ DEATH_SPASM,
+    /* 1 */ DEATH_LIMP,
+    /* 2 */ DEATH_HUNCHED
+} BossGanondrofDeathActions;
+
 void BossGanondrof_Init(Actor* thisx, GlobalContext* globalCtx);
 void BossGanondrof_Destroy(Actor* thisx, GlobalContext* globalCtx);
 void BossGanondrof_Update(Actor* thisx, GlobalContext* globalCtx);
 void BossGanondrof_Draw(Actor* thisx, GlobalContext* globalCtx);
 
-/*
+void BossGanondrof_SetupIntro(BossGanondrof* this, GlobalContext* globalCtx);
+void BossGanondrof_Intro(BossGanondrof* this, GlobalContext* globalCtx);
+void BossGanondrof_SetupPaintings(BossGanondrof* this);
+void BossGanondrof_Paintings(BossGanondrof* this, GlobalContext* globalCtx);
+void BossGanondrof_SetupNeutral(BossGanondrof* this, f32 arg1);
+void BossGanondrof_Neutral(BossGanondrof* this, GlobalContext* globalCtx);
+void BossGanondrof_SetupThrow(BossGanondrof* this, GlobalContext* globalCtx);
+void BossGanondrof_Throw(BossGanondrof* this, GlobalContext* globalCtx);
+void BossGanondrof_SetupBlock(BossGanondrof* this, GlobalContext* globalCtx);
+void BossGanondrof_Block(BossGanondrof* this, GlobalContext* globalCtx);
+void BossGanondrof_SetupReturn(BossGanondrof* this, GlobalContext* globalCtx);
+void BossGanondrof_Return(BossGanondrof* this, GlobalContext* globalCtx);
+void BossGanondrof_SetupCharge(BossGanondrof* this, GlobalContext* globalCtx);
+void BossGanondrof_Charge(BossGanondrof* this, GlobalContext* globalCtx);
+void BossGanondrof_Stunned(BossGanondrof* this, GlobalContext* globalCtx);
+void BossGanondrof_Death(BossGanondrof* this, GlobalContext* globalCtx);
+
+// extern Gfx gPhantomGanonFaceDL[];
+// extern SkeletonHeader gPhantomGanonSkel_00C710;
+// extern AnimationHeader gPhantomGanonAnim_003CA4;
```

Comment:
are all of these needed anymore?

Reply:
No, but I was using them to keep track of which things I needed to document. I'd like to get the object documented before merging if possible.

Reply:
sure yea